bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -438,32 +440,51 @@ private HomekitAccessoryType getPrimaryAccessoryType(Item item,
     private void createRootAccessories(Item item) {
         final List<Entry<HomekitAccessoryType, HomekitCharacteristicType>> accessoryTypes = HomekitAccessoryFactory
                 .getAccessoryTypes(item, metadataRegistry);
+        if (accessoryTypes.isEmpty())
+            return;
+
         final List<GroupItem> groups = HomekitAccessoryFactory.getAccessoryGroups(item, itemRegistry, metadataRegistry);
+        // Don't create accessories that are sub-accessories of other accessories
+        if (groups.stream().anyMatch(g -> !HomekitAccessoryFactory.getAccessoryTypes(g, metadataRegistry).isEmpty())) {
+            return;
+        }
+
         final @Nullable Map<String, Object> itemConfiguration = HomekitAccessoryFactory.getItemConfiguration(item,
                 metadataRegistry);
-        if (accessoryTypes.isEmpty() || !(groups.isEmpty() || groups.stream().noneMatch(g -> g.getBaseItem() == null))
-                || !itemIsForThisBridge(item, itemConfiguration)) {
+        if (!itemIsForThisBridge(item, itemConfiguration))
```

Comment:
I can't create suggestion because GitHub sees half old code, but can we also add `{ }` here?